internal/pkg/aws/acm/acm.go

```diff
@@ -66,23 +68,28 @@ func (a *ACM) ValidateCertAliases(aliases []string, certs []string) error {
 	}
 	for _, alias := range aliases {
 		if !validAliases[alias] {
-			return fmt.Errorf("%s is not a valid domain against %s", alias, strings.Join(certs, ","))
+			return &errInValidAliasAgainstCert{
+				certs:      certs,
+				alias:      alias,
+				sansOfCert: sansOfCert}
 		}
 	}
 	return nil
 }
 
-func (a *ACM) findValidAliasesAgainstCert(ctx context.Context, aliases []string, cert string) ([]string, error) {
+func (a *ACM) findValidAliasesAndAllowedDomainsAgainstCert(ctx context.Context, aliases []string, cert string) ([]string, []string, error) {
 	resp, err := a.client.DescribeCertificateWithContext(ctx, &acm.DescribeCertificateInput{
 		CertificateArn: aws.String(cert),
 	})
 	if err != nil {
-		return nil, fmt.Errorf("describe certificate %s: %w", cert, err)
+		return nil, nil, fmt.Errorf("describe certificate %s: %w", cert, err)
 	}
 	domainSet := make(map[string]bool)
+	var domainsperCert []string
```

Comment:
nit for readability:
```suggestion
	var domainsPerCert []string
```